stdlib/bytes.ml

```diff
@@ -97,19 +97,30 @@ let iter f a =
 let iteri f a =
   for i = 0 to length a - 1 do f i (unsafe_get a i) done
 
+let () = assert (3 * Sys.max_string_length <= max_int)
```

Comment:
Maybe. I still think that the decision to fix this by introducing an invariant between `Sys.max_string_length` and `max_int` that depends on the values these values take on the 32-bit platform is quite ugly and more brittle than it could be.